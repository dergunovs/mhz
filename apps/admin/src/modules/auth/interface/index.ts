export interface ILoginFormData {
  email: string;
  password: string;
}
