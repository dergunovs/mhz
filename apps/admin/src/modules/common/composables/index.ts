import { useRoute } from 'vue-router';
import { useQueryClient } from '@tanstack/vue-query';

const queryClient = useQueryClient(); // стоит проверить, вызвав invalidate

export function useInvalidate(queryKey: string) {
  const route = useRoute(); 
  function invalidate() {
    if (route.query.create === '1') {
      queryClient.invalidateQueries({ queryKey: [queryKey], exact: true });
    }
  }

  return {
    invalidate,
  };
}
