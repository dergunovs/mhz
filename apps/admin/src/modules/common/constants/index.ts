export const URL_MAIN = '/main';
export const URL_ERROR = '/404';
