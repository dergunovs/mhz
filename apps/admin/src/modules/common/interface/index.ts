import { ICategory, IManager, IManufacturer, IProduct } from 'mhz-types';

export interface ISearchResult {
  products: IProduct[];
  categories: ICategory[];
  manufacturers: IManufacturer[];
  managers: IManager[];
}
