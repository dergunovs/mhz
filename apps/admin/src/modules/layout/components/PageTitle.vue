<template>
  <div :class="$style.container">
    <h1>
      <slot></slot>
    </h1>
  </div>
</template>

<style module lang="scss">
.container {
  width: calc(100% + 64px);
  padding: 24px 32px;
  margin-top: -32px;
  margin-bottom: 32px;
  margin-left: -32px;
  color: var(--color-white);
  background: var(--gradient, linear-gradient(116deg, var(--color-primary) 0%, var(--color-success) 100%));

  h1 {
    margin: 0;
  }
}
</style>
