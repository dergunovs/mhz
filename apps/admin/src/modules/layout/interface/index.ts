import { FunctionalComponent } from 'vue';

export interface INavItem {
  _id: string;
  url: string;
  title: string;
  icon: FunctionalComponent;
}

export interface IPageTitleLink {
  url: string;
  title?: string;
}
