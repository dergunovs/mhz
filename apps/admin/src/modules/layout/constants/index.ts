import { INavItem } from '@/layout/interface';

export const NAV_ITEMS: INavItem[] = [
  { _id: 0, url: '/', title: 'Главная' },
  { _id: 1, url: '/users', title: 'Пользователи' },
  { _id: 2, url: '/shipments', title: 'Доставка' },
  { _id: 3, url: '/orders', title: 'Заказы' },
  { _id: 4, url: '/categories', title: 'Категории' },
  { _id: 5, url: '/products', title: 'Товары' },
  { _id: 5, url: '/manufacturers', title: 'Производители' },
];
