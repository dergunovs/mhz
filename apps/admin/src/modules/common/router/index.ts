import { createRouter, createWebHistory } from 'vue-router';

import { routes } from '@/common/router/routes';
import { getCookieToken } from '@/auth/composables';
import { URL_LOGIN } from '@/auth/constants';

const router = createRouter({
  history: createWebHistory(),
  routes,
  scrollBehavior(to, from, savedPosition) {
    return savedPosition ? savedPosition : { top: 0 };
  },
});

router.beforeEach((to, from, next) => {
  if (to.path !== URL_LOGIN && !getCookieToken()) {
    next({ path: `${URL_LOGIN}?logout=1` });
  } else {
    next();
  }
});

export default router;
