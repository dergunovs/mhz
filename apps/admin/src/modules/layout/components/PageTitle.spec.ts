import { DefineComponent } from 'vue';
import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';

import PageTitle from './PageTitle.vue';

import { wrapperFactory } from '@/common/test';
import { IPageTitleLink } from '@/layout/interface';

const LINKS = [
  { url: '/main', title: 'Main' },
  { url: '/about', title: 'About' },
];

const TITLE = 'Title';

const pageTitleBreadcrumbs = '[data-test="page-title-breadcrumbs"]';
const pageTitleHeader = '[data-test="page-title-header"]';

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(PageTitle, {
    props: { links: LINKS },
    slots: { default: TITLE },
  });
});

enableAutoUnmount(afterEach);

describe('PageTitle', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(PageTitle)).toBeTruthy();
  });

  it('passes props to breadcrumbs', async () => {
    expect(
      wrapper.findComponent<DefineComponent<{ links: IPageTitleLink[] }>>(pageTitleBreadcrumbs).vm.$props.links
    ).toEqual(LINKS);
  });

  it('shows breadcrumbs only if they exists', async () => {
    expect(wrapper.findComponent(pageTitleBreadcrumbs).exists()).toEqual(true);

    await wrapper.setProps({ links: undefined });

    expect(wrapper.findComponent(pageTitleBreadcrumbs).exists()).toEqual(false);
  });

  it('shows header in slot', async () => {
    expect(wrapper.find(pageTitleHeader).text()).toEqual(TITLE);
  });
});
