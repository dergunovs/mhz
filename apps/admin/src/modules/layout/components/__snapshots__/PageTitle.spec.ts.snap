// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageTitle > matches snapshot 1`] = `
"<div class="_container_9358e5">
  <ui-breadcrumbs-stub links="[object Object],[object Object]" color="white" data-test="page-title-breadcrumbs"></ui-breadcrumbs-stub>
  <h1 data-test="page-title-header">Title</h1>
</div>"
`;
